ipa-core/src/protocol/dp/mod.rs

```diff
@@ -134,75 +214,61 @@ fn d_p(success_prob: f64) -> f64 {
 /// equation (7)
 #[allow(clippy::too_many_arguments)]
 #[allow(dead_code)]
-fn epsilon_constraint(
-    num_bernoulli: u32,
-    success_prob: f64,
-    delta: f64,
-    quantization_scale: f64,
-    dimensions: f64,
-    ell_1_sensitivity: f64,
-    ell_2_sensitivity: f64,
-    ell_infty_sensitivity: f64,
-) -> f64 {
+fn epsilon_constraint(num_bernoulli: u32, noise_params: &NoiseParams) -> f64 {
```

Comment:
there are no validations for noise_params values happening here and plenty of opportunities to shoot ourselves in the foot with bad `noise_params`. I suggest you look at `NoiseParams` object and make sure it can only be constructed with values that won't lead to division by zero, overflow or bad results here